### ISSUE TITLE
Validator degrades errors to informational messages

### ISSUE THREAD

**Issue by mharthoorn** [FirelyTeam/fhir-net-api#594]
Error messages 1003 and 1022 are downgraded from error to an informational by the Validator.
They should either be skipped or stated as error.

1003: "The declared type of the element {x} is incompatible with that of the instance {y}
1022: "The stated profiles are all constraints on {x}, which is incompatible with the declared type {y} of the element

Example:
https://simplifier.net/NictizSTU3-Zib2017/Bundle-GP-Patient-Data-batch-example-01/$validate

**Comment by ahenket** [comment]
Actually... a message like this one (on that particular example):

```
Invalid : The declared type of the element (http://fhir.nl/fhir/StructureDefinition/nl-core-relatedperson) is incompatible with that of the instance ('Practitioner')

Bundle.entry[65].resource[0] Bundle.entry[1].resource[0].asserter[0]
```

Is superfluous entirely. The profile states Reference(nl-core-relatedperson | nl-core-patient | nl-core-practitioner) and the instance satisfies nl-core-practitioner. Hence any informational message on the fact that the instance does not satisfy the other two is not really helpful, but distracts.

I did not check each of the hundreds of similar messages, but I suspect a similar thing is going on with a lot of them. Please also note the warning in the same output

```
Not Supported : Slicing with a discriminator of type 'Profile' is not yet supported by this validator.

Bundle.entry[52].resource[0]
```

and

```
Not Supported : Slicing with a discriminator of type 'Type' is not yet supported by this validator.

Bundle.entry[52].resource[0].dosageInstruction[0]
```

**Comment by ewoutkramer** [comment]
Fixed by #998.
